src/index.ts

```diff
@@ -121,8 +121,40 @@ app.get("/balance/:userId", (req, res) => {
   res.json({ balances: user.balances });
 })
 
-app.get("/quote", (req, res) => {
+app.post("/quote", (req, res) => {
   // TODO: Assignment
+  const side: string = req.body.side;
+  const quantity: number = req.body.quantity;
+  let total_price: number = 0;
+  let curr_quantity = quantity;
+
+  if (side === "bid") {
+    for (let i = 0;i <= asks.length - 1;i++) {
+      if (asks[i].quantity >= curr_quantity) {
+        total_price += (asks[i].price * curr_quantity);
+        curr_quantity = 0;
+        break;
+      } else {
+        total_price += (asks[i].price * asks[i].quantity);
+        curr_quantity -= asks[i].quantity;
+      }
+    }
+  } else {
+    for (let i = 0;i <= bids.length - 1;i++) {
+      if (bids[i].quantity >= curr_quantity) {
+        total_price += (bids[i].price * curr_quantity);
+        curr_quantity = 0;
+        break;
+      } else {
+        total_price += bids[i].price * bids[i].quantity;
+        curr_quantity -= bids[i].quantity;
+      }
+    }
+  }
+
+  res.json({
+    quote: total_price
```

Comment:
If u look at the tests I think the requirement is for this to be the price of one stock

Reply:
However, in the test, the assertion was written as:

[**expect(res.body.quote).toBe(1400.9 * 2);**](https://github.com/hkirat/trade-assignment/blob/ca98b8bfe8d969ea1011cfd23dc7dc3add976ec4/src/tests/assignment.mock.ts#L38)

It appears that the expectation was for the  `total_price`  but if you prefer price of one stock  I can change it. This change would also require an adjustment in the test. 
Please let me know your preference, and I'll make the necessary updates accordingly.